src/fastfield/serializer/gcd.rs

```diff
@@ -0,0 +1,160 @@
+use fastdivide::DividerU64;
+use gcd::Gcd;
+
+pub const GCD_DEFAULT: u64 = 1;
+
+// Find GCD for iterator of numbers
+pub fn find_gcd(numbers: impl Iterator<Item = u64>) -> Option<u64> {
+    let mut numbers = numbers.filter(|n| *n != 0);
+    let mut gcd = (numbers.next()?).gcd(numbers.next()?);
+    if gcd == 1 {
```

Comment:
if you fix the above you can remove this 3 lines.

Reply:
I would keep the `gcd == 1` check, or we scan the whole fastfield when it begins with 1

Reply:
the same test exists in the loop.

Reply:
yes, but it's after the remainder check. we can have it at the beginning, but then we would check it more often than needed